libsolidity/formal/SymbolicVariables.h

```diff
@@ -52,6 +53,7 @@ class SymbolicVariable
 	virtual ~SymbolicVariable() = default;
 
 	Expression currentValue() const;
+	virtual Expression currentValueAsType(solidity::TypePointer const& _targetType) const;
```

Comment:
Not sure about the name of this... sounds a bit like you return a type that encodes the current value...
What about making ``currentValue`` virtual instead and giving it a ``solidity::TypePointer const& _targetType = TypePointer{}`` as argument? You can still return ``SymbolicVariable::currentValueAsType({});`` in the override in ``SymbolicArrayValue`` where you now return ``currentValue``...

Or alternatively there might be a better name for this...